dali/pipeline/operators/crop/bbox_crop.h

```diff
@@ -0,0 +1,251 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+#define DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+
+#include <vector>
+#include <utility>
+#include <algorithm>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+#include "dali/pipeline/operators/fused/crop_cast_permute.h"
+
+namespace dali {
+
+class BBoxCrop : public Operator<CPUBackend> {
+  static const unsigned int kAttempts = 100;
+  static const unsigned int kBboxSize = 4;
+
+ protected:
+  struct Bounds {
+    explicit Bounds(std::vector<float> &&bounds)
+        : min_(bounds[0]), max_(bounds[1]) {
+      DALI_ENFORCE(bounds.size() == 2, "Bounds should be provided as 2 values");
+      DALI_ENFORCE(min_ <= max_, "Bounds should be provided as: [min, max]");
+    }
+
+    bool Contains(float k) const { return k >= min_ && k <= max_; }
+
+    const float min_, max_;
+  };
+
+  struct Rectangle {
+    explicit Rectangle(float left, float top, float right, float bottom)
+        : left_(left),
+          top_(top),
+          right_(right),
+          bottom_(bottom),
+          area_((right - left) * (bottom - top)) {
+      DALI_ENFORCE(left_ <= right_);
+      DALI_ENFORCE(top_ <= bottom_);
+    }
+
+    bool Contains(float x, float y) const {
+      return x >= left_ && x <= right_ && y >= top_ && y <= bottom_;
+    }
+
+    float IntersectionOverUnion(const Rectangle &other) const {
+      const float x_left = std::max(left_, other.left_);
+      const float x_right = std::max(right_, other.right_);
+      const float y_top = std::min(top_, other.top_);
+      const float y_bottom = std::min(bottom_, other.bottom_);
+
+      const float intersection_area = (x_left - x_right) * (y_bottom - y_top);
+
+      return intersection_area /
+             static_cast<float>(area_ + other.area_ - intersection_area);
+    }
+
+    const float left_, top_, right_, bottom_, area_;
+  };
+
+  using Crop = Rectangle;
+  using BoundingBox = Rectangle;
+  using BoundingBoxes = std::vector<Rectangle>;
+
+ public:
+  explicit inline BBoxCrop(const OpSpec &spec)
+      : Operator<CPUBackend>(spec),
+        thresholds_{[&spec]() {
+          return spec.GetRepeatedArgument<float>("thresholds");
+        }()},
+        scaling_bounds_{[&spec]() {
+          return Bounds(spec.GetRepeatedArgument<float>("scaling"));
+        }()},
+        aspect_ratio_bounds_{[&spec]() {
+          return Bounds(spec.GetRepeatedArgument<float>("aspect_ratio"));
+        }()}
+
+  {
+    DALI_ENFORCE(!thresholds_.empty(),
+                 "At least one threshold value must be provided");
+
+    for (const auto &threshold : thresholds_) {
```

Comment:
It would be good to add DALI_ENFORCE for scaling and aspect_ratio as well

Reply:
This is provided by the constructor of the Bounds class